### ISSUE TITLE
Disconnected error, `aps_current` is not connected #244

### ISSUE THREAD

**Issue by Kacperswitalski** [aps-8id-dys/ipython-8idiuser#244]
@prjemian  L067 crashed at measurement 07555 at 3:36 am on 10/9

Got a new disconnected error: `aps_current` is not connected, so I made it into a new issue

<details>
<summary> Error from terminal </summary>

```

HDF5 workflow file name: /home/8ididata/2020-3/qzhang202010/L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1
HDF5 workflow file name: /home/8ididata/2020-3/qzhang202010/L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1
HDF5 workflow file name: /home/8ididata/2020-3/qzhang202010/L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1
HDF5 workflow file name: /home/8ididata/2020-3/qzhang202010/L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
L067_07555_CPMV_6C_7v1_NaCl4M_att00_Try1
---------------------------------------------------------------------------                                                     
DisconnectedError                         Traceback (most recent call last)
<ipython-input-22-e00c4c348150> in <module>
----> 1 _ = RE(multi_capil())

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in __call__(self, *args, **metadata_kw)
    805             self._task_fut.add_done_callback(set_blocking_event)
    806 
--> 807         self._resume_task(init_func=_build_task)
    808 
    809         if self._interrupted:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _resume_task(self, init_func)
    929                     if (exc is not None
    930                             and not isinstance(exc, _RunEnginePanic)):
--> 931                         raise exc
    932 
    933     def install_suspender(self, suspender):

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _run(self)
   1498             exit_reason = str(err)
   1499             self.log.exception("Run aborted")
-> 1500             raise err
   1501         finally:
   1502             if not exit_reason:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _run(self)
   1363                     else:
   1364                         try:
-> 1365                             msg = self._plan_stack[-1].send(resp)
   1366                         # We have exhausted the top generator
   1367                         except StopIteration:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in __call__(self, plan)
   1305         plan = monitor_during_wrapper(plan, self.monitors)
   1306         plan = baseline_wrapper(plan, self.baseline)
-> 1307         return (yield from plan)
   1308 
   1309 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in baseline_wrapper(plan, devices, name)
   1160         return (yield from plan)
   1161     else:
-> 1162         return (yield from plan_mutator(plan, insert_baseline))
   1163 
   1164 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in monitor_during_wrapper(plan, signals)
    801     plan1 = plan_mutator(plan, insert_after_open)
    802     plan2 = plan_mutator(plan1, insert_before_close)
--> 803     return (yield from plan2)
    804 
    805 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in fly_during_wrapper(plan, flyers)
    859     plan1 = plan_mutator(plan, insert_after_open)
    860     plan2 = plan_mutator(plan1, insert_before_close)
--> 861     return (yield from plan2)
    862 
    863 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

/home/beams10/8IDIUSER/bluesky_data/2020/2020-3/bp_modified_multicap.py in multi_capil()
    182     z0 = 1.3
    183 
--> 184     yield from repeat_acq(
    185         sample_name='CPMV_6C_7v1_NaCl4M',
    186         file_header='L',

/home/beams10/8IDIUSER/bluesky_data/2020/2020-3/bp_modified_multicap.py in repeat_acq(areadet, file_header, acq_rep, sample_name, Temp_list, att_value, FLY_YES_NO, samx_cen, samz_cen)
    151 
    152             # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
--> 153             yield from _one_acquire(
    154                 areadet, acq_name, ii, FLY_YES_NO, samx_cen, samz_cen)
    155             # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/home/beams10/8IDIUSER/bluesky_data/2020/2020-3/bp_modified_multicap.py in _one_acquire(areadet, acq_name, ii, FLY_YES_NO, samx_cen, samz_cen)
     53         yield from flyscan_spinner(8, 8+0.04*2.5, 0.1)
     54 
---> 55     yield from AD_Acquire(
     56         areadet,
     57         acquire_time=19.00e-6,

~/.ipython-bluesky/profile_bluesky/startup/instrument/plans/xpcs_acquire.py in AD_Acquire(areadet, file_name, acquire_time, acquire_period, num_images, path, submit_xpcs_job, atten, md)
    317 
    318     logger.info("calling full_acquire_procedure()")
--> 319     return (yield from full_acquire_procedure(md=md))

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/utils.py in dec_inner(*inner_args, **inner_kwargs)
   1112                 plan = gen_func(*inner_args, **inner_kwargs)
   1113                 plan = wrapper(plan, *args, **kwargs)
-> 1114                 return (yield from plan)
   1115             return dec_inner
   1116         return dec

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in stage_wrapper(plan, devices)
    950         return (yield from plan)
    951 
--> 952     return (yield from finalize_wrapper(inner(), unstage_devices()))
    953 
    954 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in finalize_wrapper(plan, final_plan, pause_for_debug)
    507     cleanup = True
    508     try:
--> 509         ret = yield from plan
    510     except GeneratorExit:
    511         cleanup = False

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in inner()
    948     def inner():
    949         yield from stage_devices()
--> 950         return (yield from plan)
    951 
    952     return (yield from finalize_wrapper(inner(), unstage_devices()))

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/utils.py in dec_inner(*inner_args, **inner_kwargs)
   1112                 plan = gen_func(*inner_args, **inner_kwargs)
   1113                 plan = wrapper(plan, *args, **kwargs)
-> 1114                 return (yield from plan)
   1115             return dec_inner
   1116         return dec

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in monitor_during_wrapper(plan, signals)
    801     plan1 = plan_mutator(plan, insert_after_open)
    802     plan2 = plan_mutator(plan1, insert_before_close)
--> 803     return (yield from plan2)
    804 
    805 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.ipython-bluesky/profile_bluesky/startup/instrument/plans/xpcs_acquire.py in full_acquire_procedure(md)
    285         logger.debug("before count()")
    286         # yield from bp.count([areadet], md=md)
--> 287         yield from inner_count([areadet], md=_md)
    288         logger.debug("after count()")
    289 

~/.ipython-bluesky/profile_bluesky/startup/instrument/plans/xpcs_acquire.py in inner_count(devices, md)
    259         for obj in devices:
    260             yield from bps.unstage(obj)
--> 261         yield from bps.close_run()
    262         # return ret
    263 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in close_run(exit_status, reason)
    841     :func:`bluesky.plans_stubs.open_run`
    842     """
--> 843     return (yield Msg('close_run', exit_status=exit_status, reason=reason))
    844 
    845 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in new_gen()
    793             def new_gen():
    794                 yield from ensure_generator(unmonitor_msgs)
--> 795                 yield msg
    796             return new_gen(), None
    797         else:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in new_gen()
    851                 yield from ensure_generator(complete_msgs)
    852                 yield from ensure_generator(collect_msgs)
--> 853                 yield msg
    854             return new_gen(), None
    855         else:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in new_gen()
    793             def new_gen():
    794                 yield from ensure_generator(unmonitor_msgs)
--> 795                 yield msg
    796             return new_gen(), None
    797         else:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in post_baseline()
   1149         elif msg.command == 'close_run':
   1150             def post_baseline():
-> 1151                 yield from trigger_and_read(devices, name=name)
   1152                 return (yield msg)
   1153 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in trigger_and_read(devices, name)
    909         return ret
    910     from .preprocessors import rewindable_wrapper
--> 911     return (yield from rewindable_wrapper(inner_trigger_and_read(),
    912                                           rewindable))
    913 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in rewindable_wrapper(plan, rewindable)
    691                                             restore_rewindable()))
    692     else:
--> 693         return (yield from plan)
    694 
    695 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in inner_trigger_and_read()
    903         ret = {}  # collect and return readings to give plan access to them
    904         for obj in devices:
--> 905             reading = (yield from read(obj))
    906             if reading is not None:
    907                 ret.update(reading)

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in read(obj)
     93         Msg('read', obj)
     94     """
---> 95     return (yield Msg('read', obj))
     96 
     97 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _run(self)
   1423                         # exceptions (coming in via throw) can be
   1424                         # raised
-> 1425                         new_response = await coro(msg)
   1426 
   1427                     # special case `CancelledError` and let the outer

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _read(self, msg)
   1668         obj = msg.obj
   1669         # actually _read_ the object
-> 1670         ret = obj.read(*msg.args, **msg.kwargs)
   1671 
   1672         if ret is None:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/device.py in read(self)
   1235 
   1236         for _, component in self._get_components_of_kind(Kind.normal):
-> 1237             res.update(component.read())
   1238         return res
   1239 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/utils/epics_pvs.py in wrapper(self, *args, **kwargs)
    198             return fcn(self, *args, **kwargs)
    199         else:
--> 200             raise DisconnectedError('{} is not connected'.format(self.name))
    201     return wrapper
    202 

DisconnectedError: aps_current is not connected

```

</details>

<details>
<summary> Logger </summary>

```

|2020-10-09 03:35:24.946|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:24.947|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16218, info=psutil.Process(pid=16218, name='sh', started='03:35:24')
|2020-10-09 03:35:30.150|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-732887| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07548_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07548_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:35:30.237|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07549_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:35:30.396|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:35:30.396|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07549_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:35:30.444|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:35:30.456|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:35:30.461|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-732887| - DM workflow kickoff done
|2020-10-09 03:35:30.461|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-732887| - b'id=5deaa5d2-9331-4af7-9467-d7e40347c180 owner=8idiuser status=pending startTime=1602232530.45 startTimestamp=2020/10/09 03:35:30 CDT \n'
|2020-10-09 03:35:30.506|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:35:30.604|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:35:30.737|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:30.738|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07549_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16350, info=psutil.Process(pid=16350, name='sh', started='03:35:30')
|2020-10-09 03:35:31.203|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:31.203|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16360, info=psutil.Process(pid=16360, name='sh', started='03:35:30')
|2020-10-09 03:35:36.391|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-732980| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07549_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07549_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:35:36.474|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07550_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:35:36.645|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:35:36.645|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07550_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:35:36.663|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:35:36.673|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:35:36.719|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-732980| - DM workflow kickoff done
|2020-10-09 03:35:36.719|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-732980| - b'id=8ba82d45-e7fb-4490-8d78-8bbd9f64c34f owner=8idiuser status=pending startTime=1602232536.7 startTimestamp=2020/10/09 03:35:36 CDT \n'
|2020-10-09 03:35:36.738|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:35:36.835|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:35:36.974|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:36.974|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07550_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16496, info=psutil.Process(pid=16496, name='sh', started='03:35:36')
|2020-10-09 03:35:37.436|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:37.436|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16503, info=psutil.Process(pid=16503, name='sh', started='03:35:36')
|2020-10-09 03:35:42.507|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-733073| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07550_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07550_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:35:42.576|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:35:42.746|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:35:42.747|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:35:42.767|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:35:42.778|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:35:42.839|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-733073| - DM workflow kickoff done
|2020-10-09 03:35:42.840|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-733073| - b'id=266cd074-5394-4a52-8767-3fb6c910c161 owner=8idiuser status=pending startTime=1602232542.82 startTimestamp=2020/10/09 03:35:42 CDT \n'
|2020-10-09 03:35:42.843|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:35:42.936|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:35:43.084|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:43.085|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16641, info=psutil.Process(pid=16641, name='sh', started='03:35:42')
|2020-10-09 03:35:43.552|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:43.553|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16648, info=psutil.Process(pid=16648, name='sh', started='03:35:42')
|2020-10-09 03:35:48.785|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-733166| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07551_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:35:48.849|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:35:49.048|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:35:49.048|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:35:49.078|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:35:49.088|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-733166| - DM workflow kickoff done
|2020-10-09 03:35:49.088|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-733166| - b'id=da27299b-8616-4128-8e76-d90a97bcbf7c owner=8idiuser status=pending startTime=1602232549.07 startTimestamp=2020/10/09 03:35:49 CDT \n'
|2020-10-09 03:35:49.091|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:35:49.167|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:35:49.234|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:35:49.362|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:49.362|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16793, info=psutil.Process(pid=16793, name='sh', started='03:35:48')
|2020-10-09 03:35:49.830|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:49.830|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16801, info=psutil.Process(pid=16801, name='sh', started='03:35:49')
|2020-10-09 03:35:54.862|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-733259| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07552_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:35:54.940|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:35:55.096|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:35:55.096|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:35:55.142|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:35:55.152|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:35:55.177|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-733259| - DM workflow kickoff done
|2020-10-09 03:35:55.177|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-733259| - b'id=32afe3bb-28c7-4209-b0e6-81d6690ee9e5 owner=8idiuser status=pending startTime=1602232555.16 startTimestamp=2020/10/09 03:35:55 CDT \n'
|2020-10-09 03:35:55.190|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:35:55.262|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:35:55.408|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:55.408|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16934, info=psutil.Process(pid=16934, name='sh', started='03:35:54')
|2020-10-09 03:35:55.876|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:35:55.876|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=16941, info=psutil.Process(pid=16941, name='sh', started='03:35:55')
|2020-10-09 03:36:00.986|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-733352| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07553_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:36:01.057|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:36:01.239|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:36:01.239|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:36:01.258|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:36:01.269|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:36:01.304|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:36:01.319|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-733352| - DM workflow kickoff done
|2020-10-09 03:36:01.319|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-733352| - b'id=65e74d5a-859f-41c5-8b8f-20ba79b84959 owner=8idiuser status=pending startTime=1602232561.3 startTimestamp=2020/10/09 03:36:01 CDT \n'
|2020-10-09 03:36:01.377|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:36:01.518|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:36:01.518|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=17076, info=psutil.Process(pid=17076, name='sh', started='03:36:00')
|2020-10-09 03:36:02.022|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:36:02.023|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=17083, info=psutil.Process(pid=17083, name='sh', started='03:36:01')
|2020-10-09 03:36:07.392|INFO|31139|bluesky-session|xpcs_acquire|308|Thread-733445| - DM workflow kickoff starting: analysis:True  file:/home/8ididata/2020-3/qzhang202010/L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1/L067_07554_CPMV_6C_7v1_NaCl4M_att00_Try1_0001-100000.hdf
|2020-10-09 03:36:07.465|INFO|31139|bluesky-session|bp_modified_multicap|36|bluesky-run-engine| - L067_07555_CPMV_6C_7v1_NaCl4M_att00_Try1
|2020-10-09 03:36:07.646|INFO|31139|bluesky-session|xpcs_acquire|49|bluesky-run-engine| - AD_Acquire starting
|2020-10-09 03:36:07.646|INFO|31139|bluesky-session|xpcs_acquire|60|bluesky-run-engine| - file_path = /home/8ididata/2020-3/qzhang202010/L067_07555_CPMV_6C_7v1_NaCl4M_att00_Try1/
|2020-10-09 03:36:07.670|INFO|31139|bluesky-session|xpcs_acquire|318|bluesky-run-engine| - calling full_acquire_procedure()
|2020-10-09 03:36:07.682|INFO|31139|bluesky-session|xpcs_acquire|147|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-10-09 03:36:07.744|INFO|31139|bluesky-session|xpcs_acquire|282|bluesky-run-engine| - scaler should be autocounting now
|2020-10-09 03:36:07.761|INFO|31139|bluesky-session|xpcs_acquire|313|Thread-733445| - DM workflow kickoff done
|2020-10-09 03:36:07.761|INFO|31139|bluesky-session|xpcs_acquire|314|Thread-733445| - b'id=a4867c3b-db55-473b-8e4d-77999e94075b owner=8idiuser status=pending startTime=1602232567.75 startTimestamp=2020/10/09 03:36:07 CDT \n'
|2020-10-09 03:36:07.803|INFO|31139|bluesky-session|rigaku_ufxc|104|bluesky-run-engine| - Shutter will be controlled by UFXC if shutter is left in the **showbeam** state.
|2020-10-09 03:36:07.933|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:36:07.934|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo FILE:F:L067_07555_CPMV_6C_7v1_NaCl4M_att00_Try1 | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=17220, info=psutil.Process(pid=17220, name='sh', started='03:36:07')
|2020-10-09 03:36:08.397|INFO|31139|bluesky-session|rigaku_ufxc|33|bluesky-run-engine| - process_info status: running
|2020-10-09 03:36:08.398|INFO|31139|bluesky-session|rigaku_ufxc|61|bluesky-run-engine| - command: echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 - process info: pid=17227, info=psutil.Process(pid=17227, name='sh', started='03:36:07')

```

</details>

![Screenshot 2020-10-09 081015](https://user-images.githubusercontent.com/72271116/95587301-822d7c80-0a07-11eb-811c-fef2f7c174f4.png)

**Comment by prjemian** [comment]
Good catch. We can fix that.

**Comment by prjemian** [comment]
Fix has been applied, you can reload `bp_modified_multicap.py`: `%run -m bp_modified_multicap`

The PV `S:SRcurrentAI` was not available and this stopped the scan with a `DisconnectedError`.  Because `bp_modified_multicap.py` was only looking for `StatusTimeoutError`, the plan stopped.  I expanded to `expect Exception` (which is usually too much, _you should only trap the exceptions you plan to handle_) but we can tighten that later if it becomes a problem.

Also, I'm concerned that email is not sent so I am logging the exception table just before the emails get sent.  We should test the email sending part separately.  It might be that quartz can't send email.

**Comment by Kacperswitalski** [comment]
@prjemian 
When I try to load up bluesky, I get the following error
<details>
<summary> Pythyon Terminal </summary>

```
TimeoutError: S:DesiredMode could not connect within 5.0-second timeout.

The above exception was the direct cause of the following exception:

ConnectionTimeoutError                    Traceback (most recent call last)
~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/IPython/core/shellapp.py in _exec_file(self, fname, shell_futures)
    347                     else:
    348                         # default to python, even without extension
--> 349                         self.shell.safe_execfile(full_filename,
    350                                                  self.shell.user_ns,
    351                                                  shell_futures=shell_futures,

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/IPython/core/interactiveshell.py in safe_execfile(self, fname, exit_ignore, raise_exceptions, shell_futures, *where)
   2727             try:
   2728                 glob, loc = (where + (None, ))[:2]
-> 2729                 py3compat.execfile(
   2730                     fname, glob, loc,
   2731                     self.compile if shell_futures else None)

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/IPython/utils/py3compat.py in execfile(fname, glob, loc, compiler)
    166     with open(fname, 'rb') as f:
    167         compiler = compiler or compile
--> 168         exec(compiler(f.read(), fname, 'exec'), glob, loc)
    169 
    170 # Refactor print statements in doctests.

~/.ipython-bluesky/profile_bluesky/startup/00-instrument.py in <module>
      4 """
      5 
----> 6 from instrument.collection import *
      7 
      8 # show_ophyd_symbols()

~/.ipython-bluesky/profile_bluesky/startup/instrument/collection.py in <module>
     12 
     13 from .framework import *
---> 14 from .devices import *
     15 from .plans import *
     16 from .utils import *

~/.ipython-bluesky/profile_bluesky/startup/instrument/devices/__init__.py in <module>
      4 
      5 from .aps_source import *
----> 6 from .pss import *
      7 
      8 # upstream devices

~/.ipython-bluesky/profile_bluesky/startup/instrument/devices/pss.py in <module>
     62 msg += " the bluesky session."
     63 
---> 64 if aps.inUserOperations and operations_in_8idi():
     65     suspend_I_station_status = apstools.suspenders.SuspendWhenChanged(
     66         pss.d_shutter_open_chain_A,

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/apstools/devices.py in inUserOperations(self)
    309 
    310         """
--> 311         return self.machine_status.get() in (
    312            1, "USER OPERATIONS",
    313            2, "Bm Ln Studies"

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/signal.py in get(self, as_string, timeout, connection_timeout, form, **kwargs)
   1093             as_string = self._string
   1094 
-> 1095         info = self._get_with_timeout(
   1096             self._read_pv, timeout, connection_timeout, as_string, form)
   1097 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/signal.py in _get_with_timeout(self, pv, timeout, connection_timeout, as_string, form)
   1043             self.wait_for_connection(timeout=connection_timeout)
   1044         except TimeoutError as err:
-> 1045             raise ConnectionTimeoutError(
   1046                 f"Failed to connect to {pv.pvname} "
   1047                 f"within {connection_timeout:.2} sec") from err

ConnectionTimeoutError: Failed to connect to S:DesiredMode within 5.0 sec

```

</details>

**Comment by prjemian** [comment]
On teams, I wrote:

> About 5:05 AM, I received an email that the APS Main Control Room (MCR) is experiencing problems with some of their soft IOCs.  Did not think the storage ring current signal was affected but the message was vague and it is possible.

Just confirmed that is the problem now.

```
quartz% caget S:SRcurrentAI
Channel connect timed out: 'S:SRcurrentAI' not found.
```

Since this PV (and myriad others such as `S:DesiredMode`) are unavailable, we might get a littany of problems related to the missing data needed to operate scans.  We have to wait for the MCR to resolve this problem.

**Comment by prjemian** [comment]
Next update not until noon.  Check https://status.aps.anl.gov/ for any
progress reports.

**Comment by qzhang234** [comment]
@prjemian The beam is lost again right now. However, around 5 pm, the beam was back but the ring current PV was not, so Kacper and I aligned the sample and ran a few test measurements with Spec, and it seemed like the measurements have worked. 

In case the ring current is back but the PV is not, could you disable the part in the startup script that looks for the possibly malfunctioning storage-ring-related PVs so that we may proceed with the measurement?

Thank you so much!

**Comment by qzhang234** [comment]
Update: there is still no beam but the PV is back:
```
quartz% caget S:SRcurrentAI
S:SRcurrentAI                  -0.00592948
```

**Comment by qzhang234** [comment]
This issue is believed to have been fixed with the Ophyd update. Closing the issue now.
